### ISSUE TITLE
What is the rationale for making this project AGPL?

### ISSUE THREAD

**Issue by jlumbroso** [BrightSpots/rcv#551]
I noticed the initial license for this project was Apache, but early in the commit logs this was replaced by the AGPL (I think by @tarheel at fe95903).

I am an RCV activist as well, and I am sincerely confused by the approach.

Strong copyleft licenses are activism for open-source. Here it seems like this project is both advocating for RCV but also for strong open-source, and that seems like a mix of concerns.

If your goal is to ensure that RCV comes to the US, wouldn't you welcome private vendors (who cannot be cut out of the process since a large number of election-related contracts are private) having access to this software and being able to use it themselves — this would not be great for open-source, but it would be great for RCV, as it would reduce the friction necessary for private actors to move to RCV.

I can understand someone considering that Apache is too permissive, that's why my favorite license is LGPLv3 — both it and the Mozilla Public License v2 are good trade-offs between promoting open-source but not making it impossible for private companies to use the artefacts. It states that if a vendor uses your library, they can do so without making their work open-source; but if they make any changes to your library, they must make these publicly accessible. I believe this is a good compromise between encouraging adoption and protecting the rights of the codebase.

**Comment by tarheel** [comment]
Hi @jlumbroso, thanks for the question! You make good points. Our original decision to adopt AGPL was in response to strong encouragement from open-source proponents, as you may have guessed. You're right that a license like LGPL would make it much easier for a vendor to incorporate our code into their products. In practice, though, we've found that it's unlikely that a vendor would want to engage with our work in this way; they're more likely to a) merely look to our code as inspiration for their own proprietary implementation, or b) keep their focus on the voting machines and cast vote record generation and let our software handle the RCV tabulation directly (as is now happening in multiple jurisdictions). If we did encounter a situation in which a vendor wanted to pull our work into their own codebase and we saw potentially significant benefit for the broader RCV movement from enabling that adoption, we would give serious consideration to modifying our license. But, as far as we're aware, this is only a hypothetical at the moment.

**Comment by HEdingfield** [comment]
Since we're going to be modifying the license in the near future (CC: @chughes297, @moldover), let's also make sure to update all the copyright blocks to say 2022 (they currently say 2020) and fix [this typo](https://github.com/BrightSpots/rcv/pull/603/commits/ece5a579d9aafefb290e751a81b467e2079826be) where it says "Universal RCTab" instead of "RCTab".

**Comment by moldover** [comment]
Good call 👍 
So @chughes297 we have the the current copyright header text at the top of every source code file:

/*
 * RCTab
 * Copyright (c) 2017-2020 Bright Spots Developers.
 *
 * This program is free software: you can redistribute it and/or modify it under the terms of the
 * GNU Affero General Public License as published by the Free Software Foundation, either version 3
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
 * the GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License along with this
 * program.  If not, see <http://www.gnu.org/licenses/>.
 */

And we need to know what the new text should be.

**Comment by moldover** [comment]
This is from an email from @chughes297 , just to keep @jlumbroso and everyone up to date:

Hi Hylton, Jon, Louis, and Tamir, 

Hope you're all doing well. 

This email is a brief update on our licensing strategy for RCTab. TL;DR: we are moving RCTab entirely over to an [MPL 2.0](https://fossa.com/blog/open-source-software-licenses-101-mozilla-public-license-2-0/) license and dropping AGPL entirely. 

Rene and I spoke with Heather and Greg at OSET and they advised us that the most realistic path forward is to offer RCTab under a single license - namely, MPL 2.0.

I assumed it was our best strategy to offer RCTab under AGPL via GitHub/to non-commercial licensees and under MPL 2.0 for commercial licensees. Heather explained, though, that once you provide the software under MPL to one person you need to provide that same software version under MPL to all potential users. 

I also thought keeping the GitHub version under an AGPL license would give us more bargaining power with the vendors. Both Greg and Heather thought this wouldn't really pan out - because one vendor (Hart) has already accepted RCTab under the MPL, that sets the stage for other vendors to accept the same license. We may need to be ready to educate the vendors to get them on board with MPL, but both Greg and Heather assured me we wouldn't lose anything in a negotiation with the vendors if we're already providing the software under MPL 2.0. It will also be administratively easier to just provide a single license and not track which users are using the software under which license. 

To sum up: we're just going to provide RCTab under a single license - MPL 2.0. We are dropping AGPL. 

One aside that I want to note: 

MPL 2.0 does also allow you to license derivative works under AGPL or other more-copyleft licenses, too, so we could still provide versions of RCTab under more aggressive licenses (if we want to). 

Two things I need you guys to do to officially move us over to MPL 2.0: 

1) Run a BLAME function on the RCTab code to double-check that we've captured all contributors to the software on the license wavier
2) Update the license on GitHub to MPL 2.0 (from here: https://www.mozilla.org/en-US/MPL/2.0/). 

Let me know if you have any questions. 

Thanks!

**Comment by moldover** [comment]
So @chughes297 we have the the current copyright header text at the top of every source code file, and we need to know what the new text should be:

/*

    RCTab
    Copyright (c) 2017-2020 Bright Spots Developers.
    This program is free software: you can redistribute it and/or modify it under the terms of the
    GNU Affero General Public License as published by the Free Software Foundation, either version 3
    of the License, or (at your option) any later version.
    This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
    without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See
    the GNU Affero General Public License for more details.
    You should have received a copy of the GNU Affero General Public License along with this
    program. If not, see http://www.gnu.org/licenses/.
    */

**Comment by jlumbroso** [comment]
@moldover @HEdingfield I believe this is an excellent decision, and in fact one that might make a difference for the better!!

Well done all of you! 👏🏻👏🏻👏🏻👏🏻

@chughes297 Let me know if you would like help with the headers.

**Comment by rrojas350** [comment]
@moldover I believe an easy fix would be to replace GNU Affero General Public License with Mozilla Public License V2.0, just find and replace. Something like this:

RCTab
Copyright (c) 2017-2020 Bright Spots Developers.
This program is free software: you can redistribute it and/or modify it under the terms of the
Mozilla Public License Version 2.0 as published by the Mozilla Foundation.
This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See
the Mozilla Public License Version 2.0 for more details.
You should have received a copy of the Mozilla Public License Version 2.0 along with this
program. If not, see https://www.mozilla.org/en-US/MPL/2.0/.

@chughes297 will have to chime in if that still covers us from a legal standpoint.

**Comment by moldover** [comment]
@rrojas350 the current copyright text is specific to GNU: https://www.gnu.org/licenses/gpl-howto.html  
An ad-hoc find/replace does not seem like a good idea to me.

Can you please follow up with @chughes297 IRL and make sure he is looped into Github conversations, and find out what, if any, new copyright text should go into the source header files?

**Comment by tarheel** [comment]
According to his email auto-response, Chris was out of the office for a while but should be back now.

**Comment by rrojas350** [comment]
Maybe I should preface my comments with, "this does not constitute legal advice."

Chris just got back today and is catching up on emails, so I'm sure he'll chime in soon. I'll shoot him a text just to verify.

**Comment by moldover** [comment]
I found these instructions: https://www.mozilla.org/en-US/MPL/headers/ 
They have a similar setup and go into the source code headers so it could look like this:

/*
 * RCTab
 * Copyright (c) 2017-2022 Bright Spots Developers.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 */

**Comment by chughes297** [comment]
Sorry for my delay here - was closing out a few other projects earlier this week! This looks right to me.

**Comment by moldover** [comment]
No worries!  I got the sources all update I think so it should be gg.

**Comment by moldover** [comment]
closed via #604
